### PR TITLE
Added section for iSCSI multipath support

### PR DESCRIPTION
**Pull request by gaurav-nelson**
For https://trello.com/c/C1mn7BoP/467-document-add-multipath-support-for-iscsi-volume-plugin?menu=filter&filter=@gauravnelson

- Fixed passive vocie
- Added section for multipath support

### CODE REVIEW

**Comment by gaurav-nelson** [comment]
[Document: Add multipath support for iscsi volume plugin.](https://trello.com/c/C1mn7BoP/467-document-add-multipath-support-for-iscsi-volume-plugin)

File: install_config/persistent_storage/persistent_storage_iscsi.adoc [hunk 1]
```diff
@@ -15,7 +15,7 @@ toc::[]
 You can provision your {product-title} cluster with
 xref:../../architecture/additional_concepts/storage.adoc#architecture-additional-concepts-storage[persistent storage] using
 https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/Storage_Administration_Guide/ch-iscsi.html[iSCSI].
-Some familiarity with Kubernetes and iSCSI is assumed.
+For the purpose of this document we assume some famalirity with Kubernetes and iSCSI.
```
Comment (humblec) [inline install_config/persistent_storage/persistent_storage_iscsi.adoc]: typo -> "famalirity"

File: install_config/persistent_storage/persistent_storage_iscsi.adoc [hunk 2]
```diff
@@ -84,3 +83,37 @@ namespace. Any attempt to access a persistent volume across a namespace causes
 the pod to fail.
 
 Each iSCSI LUN must be accessible by all nodes in the cluster.
+
+[[iscsi-multipath]]
+
+=== iSCSI Multipathing
+For iSCSI based storage, you can configure multiple paths by using same IQN for more than one IP addresses. Multipathing ensures access to the persistent volume when one or more of the components in a path fail.
```
Comment (humblec) [inline install_config/persistent_storage/persistent_storage_iscsi.adoc]: .....for more than on Target portal IP addresses....

File: install_config/persistent_storage/persistent_storage_iscsi.adoc [hunk 3]
```diff
@@ -84,3 +83,37 @@ namespace. Any attempt to access a persistent volume across a namespace causes
...
+      name: iscsipd-rw
+  volumes:
+  - name: iscsipd-rw
+    iscsi:
+      targetPortal: 10.0.2.15:3260
+      portals: ['10.0.2.16:3260', '10.0.2.17:3260', '10.0.2.18:3260'] <1>
+      iqn: iqn.2016-04.test.com:storage.target00
+      lun: 0
+      fsType: ext4
+      readOnly: true
```
Comment (humblec) [inline install_config/persistent_storage/persistent_storage_iscsi.adoc]: I think mentioning 'persistent Volume spec' is preferred  than 'spec' directly in pod spec. @rootfs any thoughts.

**Comment by humblec** [comment]
@rootfs PTAL.

File: install_config/persistent_storage/persistent_storage_iscsi.adoc (hunk 3 above)
Comment (rootfs) [inline install_config/persistent_storage/persistent_storage_iscsi.adoc]: yes, if you can supply a PV spec, that'll be more generic.

Comment (gaurav-nelson) [inline install_config/persistent_storage/persistent_storage_iscsi.adoc]: @humblec can you please provide the example I can use here?

**Comment by gaurav-nelson** [comment]
@humblec can you provide an example I can use here

**Comment by rootfs** [comment]
@gaurav-nelson here is an example
```yaml
apiVersion: v1
  kind: PersistentVolume
  metadata:
    name: iscsi_pv
  spec:
    capacity:
      storage: 1Gi
    accessModes:
      - ReadWriteOnce
    iscsi:
      targetPortal: 10.0.0.1:3260
      iqn: iqn.2016-04.test.com:storage.target00
      lun: 0
      fsType: ext4
      readOnly: false
```

**Comment by gaurav-nelson** [comment]
@bfallonf peer review please

File: install_config/persistent_storage/persistent_storage_iscsi.adoc [hunk 4]
```diff
@@ -15,7 +15,7 @@ toc::[]
 You can provision your {product-title} cluster with
 xref:../../architecture/additional_concepts/storage.adoc#architecture-additional-concepts-storage[persistent storage] using
 https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/Storage_Administration_Guide/ch-iscsi.html[iSCSI].
-Some familiarity with Kubernetes and iSCSI is assumed.
+For the purpose of this document we assume some familiarity with Kubernetes and iSCSI.
```
Comment (bfallonf) [inline install_config/persistent_storage/persistent_storage_iscsi.adoc]: "we" is first person and advised against. I'm guessing that's why this was passive int he first place. Maybe: 

"For the purpose of this document, Red Hat assumes familiarity with..."

File: install_config/persistent_storage/persistent_storage_iscsi.adoc [hunk 5]
```diff
@@ -31,9 +31,8 @@ storage provider.
 [[provisioning-iscsi]]
 
 == Provisioning
-Storage must exist in the underlying infrastructure before it can be mounted as
-a volume in {product-title}. All that is required for iSCSI is iSCSI target
-portal, valid iSCSI IQN, valid LUN number, and filesystem type, and the
+Verify that the storage exists in the underlying infrastructure before mounting it as a volume in {product-title}. All that is required for iSCSI is iSCSI target
+portal, valid iSCSI Qualified Name (IQN), valid LUN number, and filesystem type, and the
```
Comment (bfallonf) [inline install_config/persistent_storage/persistent_storage_iscsi.adoc]: I think there's "the" and "and" things above? Maybe:

"All that is required for iSCSI is the iSCSI target portal, a valid iSCSI Qualified Name (IQN), a valid LUN number, the filesystem type, and the `*PersistentVolume*` API."

?

Comment (bfallonf) [inline install_config/persistent_storage/persistent_storage_iscsi.adoc]: EDIT: hmm unless semi-colons will help divide the parts of the list more?

File: install_config/persistent_storage/persistent_storage_iscsi.adoc [hunk 6]
```diff
@@ -84,3 +83,33 @@ namespace. Any attempt to access a persistent volume across a namespace causes
 the pod to fail.
 
 Each iSCSI LUN must be accessible by all nodes in the cluster.
+
+[[iscsi-multipath]]
+
+=== iSCSI Multipathing
+For iSCSI based storage, you can configure multiple paths by using same IQN for more than one target portal IP addresses. Multipathing ensures access to the persistent volume when one or more of the components in a path fail.
```
Comment (bfallonf) [inline install_config/persistent_storage/persistent_storage_iscsi.adoc]: s/iSCSI based/iSCSI-based

s/by using same/by using the same

s/IP addresses/IP address

**Comment by bfallonf** [comment]
@gaurav-nelson A few nits. Looks good otherwise.

File: install_config/persistent_storage/persistent_storage_iscsi.adoc (hunk 4 above)
Comment (gaurav-nelson) [inline install_config/persistent_storage/persistent_storage_iscsi.adoc]: I think I'll change it back to the passive from thanks :)

**Comment by gaurav-nelson** [comment]
[rev_history]
|xref:../install_config/persistent_storage/persistent_storage_iscsi.adoc#install-config-persistent-storage-persistent-storage-iscsi[iSCSI Multipathing]
|Added a new topic for multipath support for iSCSI volume plugin.
%

**Pull request merged**